### PR TITLE
refactor: refine data structure rendering TimeGrid

### PR DESCRIPTION
**Pull request by adhrinae**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

- Implemented `createTimeGridData` function to pass props in order to render the `TimeGrid` component in efficient. 
- Changed affected components with new props.
- Polish stories of `column`, `timegrid`.

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/calendar/src/helpers/__tests__/grid.spec.ts [hunk 1]
```diff
@@ -181,3 +186,154 @@ describe('createDateMatrixOfMonth', () => {
     expect(resultStartFromFriday).toEqual(expectStartFromFriday);
   });
 });
+
+describe('getColumnStyles', () => {
+  it('should create default styles of a week', () => {
+    // Given
+    const weekDates = getWeekDates(new TZDate('2021-01-28T00:00:00'), {
+      startDayOfWeek: 0,
```
Comment (lja1018) [inline apps/calendar/src/helpers/__tests__/grid.spec.ts:194]:
```suggestion
      startDayOfWeek: Day.SUN,
```
은 어떠신가요?

File: apps/calendar/src/helpers/grid.ts [hunk 2]
```diff
@@ -24,8 +24,10 @@ import {
   Matrix3d,
   TimeGridEventMatrix,
 } from '@t/events';
+import { CommonGridColumn, TimeGridData, TimeGridRow } from '@t/grid';
```
Comment (lja1018) [inline apps/calendar/src/helpers/grid.ts:27]: 여기도 unused import 있습니다!

File: apps/calendar/src/helpers/grid.ts [hunk 3]
```diff
@@ -345,3 +347,87 @@ export function createDateMatrixOfMonth(
     }, [] as TZDate[])
   );
 }
+
+export function getWeekDates(
+  renderDate: TZDate,
+  { startDayOfWeek = 0, workweek }: WeekOptions
```
Comment (lja1018) [inline apps/calendar/src/helpers/grid.ts:353]:
```suggestion
  { startDayOfWeek = Day.SUN, workweek }: WeekOptions
```
은 어떠신가요?

File: apps/calendar/src/helpers/grid.ts [hunk 4]
```diff
@@ -345,3 +347,87 @@ export function createDateMatrixOfMonth(
...
+    .reduce((result, currentDateAndWidth, index) => {
+      const prev = result[index - 1];
+
+      result.push({
+        ...currentDateAndWidth,
+        left: index === 0 ? 0 : prev.left + prev.width,
+      });
+
+      return result;
+    }, [] as CommonGridColumn[]);
```
Comment (lja1018) [inline apps/calendar/src/helpers/grid.ts:403]:
```suggestion
    .reduce<CommonGridColumn[]>((result, currentDateAndWidth, index) => {
      const prev = result[index - 1];

      result.push({
        ...currentDateAndWidth,
        left: index === 0 ? 0 : prev.left + prev.width,
      });

      return result;
    }, []);
```
는 어떠신가요?

File: apps/calendar/src/helpers/grid.ts [hunk 5]
```diff
@@ -345,3 +347,87 @@ export function createDateMatrixOfMonth(
...
+    hourEnd: number;
+  }
+): TimeGridData {
+  const columns = getColumnsData(datesOfWeek);
+
+  const steps = options.hourEnd - options.hourStart;
+  const baseHeight = 100 / steps;
+  const rows = range(options.hourStart, options.hourEnd).map((hour, index) => {
+    const startTime = `${hour}:00`.padStart(5, '0') as FormattedTimeString;
+    const endTime = `${hour + 1}:00`.padStart(5, '0') as FormattedTimeString;
```
Comment (lja1018) [inline apps/calendar/src/helpers/grid.ts:419]: start와 end time이 `00:00`부터 `99:99`로 추론되네요 ㅋㅋ..
십의자리만이라도 0,1,2로 제한하는건 어떠신가요?

File: apps/calendar/stories/timegrid.stories.tsx [hunk 6]
```diff
@@ -80,32 +83,45 @@ function getEvents() {
...
+  return createTimeGridData(weekDates, { hourStart: 0, hourEnd: 24 });
+}
 
-  return (
-    <ProviderWrapper>
-      <TimeGrid events={events} />
-    </ProviderWrapper>
-  );
+type TimeGridProps = ComponentProps<typeof TimeGrid>;
+const Template: StoryFn<TimeGridProps> = (args: ComponentProps<typeof TimeGrid>) => (
```
Comment (lja1018) [inline apps/calendar/stories/timegrid.stories.tsx:93]:
```suggestion
const Template: StoryFn<TimeGridProps> = (args) => (
```
가 되야겠네요

File: apps/calendar/types/time/datetime.d.ts [hunk 7]
```diff
@@ -18,3 +18,6 @@ interface CellStyle {
 interface CellInfo extends CellStyle {
   date: TZDate;
 }
+
+type Digit = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
+export type FormattedTimeString = `${Digit}${Digit}:00:00`;
```
Comment (lja1018) [inline apps/calendar/types/time/datetime.d.ts:23]: 위 리뷰대로 십의자리만이라도 0,1,2로 제한하는건 어떠신가요?

**Review by lja1018 — APPROVED** [review]
리뷰 완료합니다. 고생하셨습니다!

File: apps/calendar/src/helpers/grid.ts (hunk 5 above)
Comment (adhrinae) [reply apps/calendar/src/helpers/grid.ts:419]: 미처 생각못했네요. 좋은 제안이에요.

**Comment by adhrinae** [comment]
급하게 마무리하다 빠뜨린게 좀 있었네요. 어여 보충해야겠어요.

**Pull request merged**